raiden/api/validations/light_client_balance.py

```diff
@@ -0,0 +1,33 @@
+from http import HTTPStatus
+
+import structlog
+from flask import request
+
+from raiden.api.validations.api_error_builder import ApiErrorBuilder
+from raiden.api.validations.light_client_authorization import get_key_error
+from raiden.lightclient.handlers.light_client_service import LightClientService
+
+log = structlog.get_logger(__name__)
+
+
+def requires_lc_balance(func):
+    """
+        Check if the endpoint handled by func was called with a Lc that has balance greater than 0.
+    """
+    def inner(*args, **kwargs):
+        raiden_api = args[0].raiden_api
+        key_error = get_key_error(raiden_api.raiden.wal)
```

Comment:
im not sure about checking the api key here.

i thought that those rest endpoints were checking the api key already, but seems that only the payment messages ones were controled. 

anyway, i found more clear to split this into two decorators

- requires_api_key
- requires_lc_balance

then on the open_light you can just put the two decorators on that order

what do u think?

Reply:
I tried that but you need to depend on requires_api_key and maybe the user doesn't use that in the call and you can have a problem there.